galley/pkg/config/analysis/analyzers/testdata/virtualservice_destinationhosts.yaml

```diff
@@ -60,3 +69,25 @@ spec:
     - destination:
         host: external-reviews.org  # Referring to a ServiceEntry host is valid and should not generate an error
                                     # Since this is an "external" service, subset is omitted
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: reviews-bookinfo-eu
+  namespace: default
+spec:
+  http:
+  - route:
+    - destination:
+        host: reviews.eu.bookinfo.com # Should match *.eu.bookinfo.com
```

Comment:
Please also add a test case where the virtualservice destination host is wildcarded.

Also, a test case where the service entry is in a different namespace and NOT exported. (Not directly related to your change, but it seems like something we should make sure we cover).

Reply:
Isn't the last test case a VS wildcard matching? Or did you mean something else than exact match wildcard?

Reply:
What I had in mind was something like this:
```yaml
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: service-entry-bookinfo-unexported
  namespace: other
spec:
  exportTo:
    - "."
  hosts:
  - "unexported.bookinfo.com" 
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: reviews-bookinfo-eu-wildcard
  namespace: default
spec:
  http:
  - route:
    - destination:
        # Should generate an error since although the host exists, it's in another namespace and unexported
        host: "unexported.bookinfo.com"
```

Reply:
Ouh,  I see. I solved that approach with the wildcard matching by making * host unexported, otherwise it should match all those host queries and no negatives should be possible. But I can make exact match test also.